src/bitops.c

```diff
@@ -710,59 +733,140 @@ void bitopCommand(client *c) {
-                    minlen -= sizeof(unsigned long)*4;
+                    lres += 4;
+                    j += step;
+                    minlen -= step;
                 }
             } else if (op == BITOP_NOT) {
-                while(minlen >= sizeof(unsigned long)*4) {
+                while(minlen >= step) {
                     lres[0] = ~lres[0];
                     lres[1] = ~lres[1];
                     lres[2] = ~lres[2];
                     lres[3] = ~lres[3];
-                    lres+=4;
-                    j += sizeof(unsigned long)*4;
-                    minlen -= sizeof(unsigned long)*4;
+                    lres += 4;
+                    j += step;
+                    minlen -= step;
+                }
+            } else if (op == BITOP_DIFF || op == BITOP_DIFF1 || op == BITOP_ANDOR) {
+                while(minlen >= step) {
+                    for (i = 1; i < numkeys; i++) {
+                        lres[0] |= lp[i][0];
+                        lres[1] |= lp[i][1];
+                        lres[2] |= lp[i][2];
+                        lres[3] |= lp[i][3];
+                        lp[i] += 4;
+                    }
+                    lres += 4;
+                    j += step;
+                    minlen -= step;
+                    processed += step;
+                }
+
+                lres = (unsigned long*) res;
+                switch (op) {
+                case BITOP_DIFF:
+                    for (i = 0; i < processed; i += step) {
+                        lres[0] = (first_key[0] & ~lres[0]);
+                        lres[1] = (first_key[1] & ~lres[1]);
+                        lres[2] = (first_key[2] & ~lres[2]);
+                        lres[3] = (first_key[3] & ~lres[3]);
+                        lres += 4;
+                        first_key += 4;
+                    }
+                    break;
+                case BITOP_DIFF1:
+                    for (i = 0; i < processed; i += step) {
+                        lres[0] = (~first_key[0] & lres[0]);
+                        lres[1] = (~first_key[1] & lres[1]);
+                        lres[2] = (~first_key[2] & lres[2]);
+                        lres[3] = (~first_key[3] & lres[3]);
+                        lres += 4;
+                        first_key += 4;
+                    }
+                    break;
+                case BITOP_ANDOR:
+                    for (i = 0; i < processed; i += step) {
+                        lres[0] = (first_key[0] & lres[0]);
+                        lres[1] = (first_key[1] & lres[1]);
+                        lres[2] = (first_key[2] & lres[2]);
+                        lres[3] = (first_key[3] & lres[3]);
+                        lres += 4;
+                        first_key += 4;
+                    }
+                    break;
+                }
+            } else if (op == BITOP_ONE) {
+                while(minlen >= step) {
+                    memset(lcommon_bits, 0, sizeof(unsigned long)*4);
```

Comment:
```suggestion
                    memset(lcommon_bits, 0, sizeof(lcommon_bits));
```
Perhaps this is a safer alternative as we don't need to track byte count.